clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/model/SecurityGroupLookupFactory.groovy

```diff
@@ -70,28 +72,31 @@ class SecurityGroupLookupFactory {
       new SecurityGroupUpdater(newSecurityGroup, amazonEC2)
     }
 
-    SecurityGroupUpdater getSecurityGroupByName(String accountName, String name, String vpcId, boolean skipCache = false) {
+    SecurityGroupUpdater getSecurityGroupByName(String accountName, String name, String vpcId) {
       final credentials = getCredentialsForName(accountName)
       if (!credentials) { return null }
-      final amazonEC2 = amazonClientProvider.getAmazonEC2(credentials, region, skipCache)
-      final securityGroup = getSecurityGroups(accountName, amazonEC2, skipCache).find {
+
+      def amazonEC2 = amazonClientProvider.getAmazonEC2(credentials, region, true)
+      def cachedSecurityGroupKey = name.toLowerCase() + "." + vpcId
+      def cachedSecurityGroup = securityGroupByName.get(cachedSecurityGroupKey)
+      if (cachedSecurityGroup) {
+        return new SecurityGroupUpdater(cachedSecurityGroup, amazonEC2)
+      }
+
+      def describeSecurityGroupsRequest = new DescribeSecurityGroupsRequest().withFilters(
```

Comment:
Verified that this results in only a single call to AWS (ie. it doesn't fetch the world)